### PR TITLE
Adding 'program-editor' role and updating other permissions

### PR DESCRIPTION
**Pull request by halloffame**
This person has access to manage all existing programs, subjects, and generic objects (but not create new ones). Eventually we should probably make those objects be permissions subjects so we can scope the ones people see by the ones they should be editing.

Changed the history tab to only show up if you `can_manage?(:activity_logs)`. This is what it is looking at anyways to render the page.

Also fixing the navigation to only show links to index pages you have access to.

### CODE REVIEW

File: app/extensions/user.rb [hunk 1]
```diff
@@ -0,0 +1,7 @@
+User.class_eval do
+
+  def program_editor?
```
Comment (mikestephens) [inline app/extensions/user.rb]: I've had many a discussion about this type of thing with Jeff. This is policy related therefore should be put in the policy files and not in a central location like a user extension.

File: app/policies/academic_subject_policy.rb [hunk 2]
```diff
@@ -1,16 +1,18 @@
 class AcademicSubjectPolicy < ApplicationPolicy
   class Scope < Struct.new(:user, :scope)
     def resolve
-      if user.admin?
-        scope.all
-      else
-        scope.none
-      end
+      scope.all
```
Comment (mikestephens) [inline app/policies/academic_subject_policy.rb]: Is this best to leave as scope.all for anyone?

File: app/policies/generic_object_policy.rb [hunk 3]
```diff
@@ -1,16 +1,18 @@
 class GenericObjectPolicy < ApplicationPolicy
   class Scope < Struct.new(:user, :scope)
     def resolve
-      if user.admin?
-        scope.all
-      else
-        scope.none
-      end
+      scope.all
```
Comment (mikestephens) [inline app/policies/generic_object_policy.rb]: Same thing here

File: config/initializers/model_extensions.rb [hunk 4]
```diff
@@ -0,0 +1,3 @@
+Dir[Rails.root.join("app", "extensions", "*.rb")].each do |path|
```
Comment (mikestephens) [inline config/initializers/model_extensions.rb]: Don't forget to remove this file once you remove the user extension.

**Comment by mikestephens** [comment]
+1 good to be merged whenever the tests pass

File: app/policies/academic_subject_policy.rb (hunk 2 above)
Comment (halloffame) [inline app/policies/academic_subject_policy.rb]: Yeah, this is fine for now. We have to explicitly give people `program_editor` access right now anyways, so we know who will have access to this.

**Pull request merged**